crates/typst/src/eval/call.rs

```diff
@@ -368,6 +261,111 @@ pub(crate) fn call_closure(
     Ok(output)
 }
 
+/// Method calls have some special cases which we convert to values eagerly. Otherwise
+/// this contains the evaluated method to call and the arguments to call with.
+enum MethodCall {
```

Comment:
It's not really always a `MethodCall`. It could be a function call on a field (e.g. `table.cell`) and then the special case in particular can also be other stuff.

How about `FieldCall` and renaming `Special` to `Resolved` since the call is already resolved?